src/NosCore.GameObject/Providers/MinilandProvider/MinilandProvider.cs

```diff
@@ -155,9 +159,21 @@ public void SetState(long characterId, MinilandState state)
                 var ml = _minilandIds[characterId];
                 var miniland = _mapInstanceProvider.GetMapInstance(ml.MapInstanceId);
                 ml.State = state;
+
                 if (ml.State != MinilandState.Open)
                 {
-                    miniland.Kick(o => o.VisualId != characterId);
+                    if (ml.State == MinilandState.Private)
+                    {
+                        List<long> friends = _friendHttpClient.GetListFriends(characterId)
+                            .Where(w => w.RelationType != CharacterRelationType.Blocked)
```

Comment:
should be by default as friend and not blacklist

Reply:
Hum, on CharacterEntityEntension, method GenerateFinit() line 148, we need to load entire FriendList no ? For load blocked relation on client.
I could by default said non blocked, and do the Where clause on GetListFriends method and add a boolean on it called "includeBlocked" by default at false, and apply Where clause on it ? Like that ?
`List<CharacterRelationStatus> GetListFriends(long visualEntityVisualId, boolean includeBlocked = false) {
    includeBlocked ? return Get<List<CharacterRelationStatus>>(visualEntityVisualId) : return Get<List<CharacterRelationStatus>>(visualEntityVisualId).Where(w => w.RelationType != CharacterRelationType.Blocked);
}`

Reply:
the service never return blacklist in friends

Reply:
Okay, i change it so ;) thx for reviewed it.